### DISCUSSION TITLE
AMD FSR support

### DISCUSSION THREAD

**Discussion by FugaziScript** [obsproject/obs-studio#5051] [Development Questions] [answered]
Hello everyone, 
I have a suggestion to add amd fsr support as an upscaling method in the obs software. I think it would be useful to some people (for example: gaming in 1080p and streaming in 1440p).
What do you think, is it possible?

**Comment by jp9000** [comment]
I honestly doubt it'd be all that useful. Most people downscale, not upscale. That's just my opinion though.

**Comment by skeletonbow** [comment, marked as answer]
> Hello everyone,
> I have a suggestion to add amd fsr support as an upscaling method in the obs software. I think it would be useful to some people (for example: gaming in 1080p and streaming in 1440p).
> What do you think, is it possible?

AMD FSR is open source under an MIT style license, which is compatible with the GNU GPL license so it is certainly possible to do this if someone is interested in doing the work.  Game developers have stated it can be implemented in their games in as little as a few hours time, or up to a few days depending on various factors.  So it should likely be possible to implement it inside of OBS within a few hours to days also if anyone is curious enough to try it out and see if there might be useful use cases within OBS.

Since it seems quite easy to implement from anecdotal evidence out there, someone will most likely convert the shader code to be loadable in one of the OBS shader plugins, or write a standalone plugin for this at some point.  I believe Xaymar (the author of the StreamFX plugin) has been experimenting with upscaling technologies within OBS so I speculate it might be possible to see FSR show up experimentally in a future StreamFX plugin update at some point, although that is personal speculation.

I haven't looked at AMD's code yet to see what all is involved, but I'm curious about it as well.  If it is something that would be possible to load as a pixel shader in the existing shader plugins, I'd be inclined to experiment with it myself if someone else doesn't beat me to it.  If it is more elaborate than that or requires being a standalone plugin for some reason, I'm sure someone will try it at some point.

I imagine that the pathway to seeing FSR supported in OBS, would be someone making it available as a plugin or shader that can be loaded into OBS unofficially first, and proving it's usefulness for particular use cases.  If that turns out to be true, then it's a matter of either just using the plugin, or convincing the devs to integrate support natively in the future based on proven usefulness to the average user for specific use cases.

So it's possible, but unproven as to whether it is useful in the context of OBS at this point.  Something worthy for someone motivated enough to consider experimenting with though.

**Comment by pegasusearl** [comment]
As for use case, it's useful for streaming old games, especially the 2D non pixel art ones. As for playing itself, there is already program called [Magpie](https://github.com/Blinue/Magpie) and [Lossless Scaling](https://store.steampowered.com/app/993090/Lossless_Scaling) to upscale these old games with FSRCNNX.

Unfortunately OBS cannot capture the upscaled content if we use Window Capture, they have to use Display Capture instead. But if someone was streaming, they won't be using Display Capture as they don't want their viewers to see other things.
